tests/metrics/test_cuboid_metrics.py

```diff
@@ -8,4 +17,300 @@
     )
 
-# TODO(gunnar): Add Cuboid tests!
+CAR_LABEL = "car"
+PEDESTRIAN_LABEL = "pedestrian"
+
+
+def test_cuboid_metrics_simple():
+    # single item, perfect predictions
+    annotations = [
+        CuboidAnnotation(
+            label=CAR_LABEL,
+            position=Point3D(0, 0, 0),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+        CuboidAnnotation(
+            label=CAR_LABEL,
+            position=Point3D(1000, 1000, 1000),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+    ]
+    predictions = [
+        CuboidPrediction(
+            label=CAR_LABEL,
+            position=Point3D(0, 0, 0),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+        CuboidPrediction(
+            label=CAR_LABEL,
+            position=Point3D(1000, 1000, 1000),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+    ]
+    assert CuboidIOU().eval(annotations, predictions) == ScalarResult(
+        1.0, len(annotations)
+    ), "Unexpected Cuboid IoU result"
+    assert CuboidPrecision().eval(annotations, predictions) == ScalarResult(
+        1.0, len(annotations)
+    ), "Unexpected Cuboid Precision result"
+    assert CuboidRecall().eval(annotations, predictions) == ScalarResult(
+        1.0, len(annotations)
+    ), "Unexpected Cuboid Recall result"
+
+
+def test_cuboid_metrics_numerical_check():
+    # single item, realistic predictions w/ matches and non-matches
+    annotations = [
+        CuboidAnnotation(
+            label=CAR_LABEL,
+            position=Point3D(0, 0, 0),
+            dimensions=Point3D(10, 10, 5),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+        CuboidAnnotation(
+            label=CAR_LABEL,
+            position=Point3D(1000, 1000, 1000),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+        CuboidAnnotation(
+            label=CAR_LABEL,
+            position=Point3D(-100, -100, -100),
+            dimensions=Point3D(10, 10, 10),
+            yaw=0.0,
+            reference_id="item_A",
+        ),  # false negative
+    ]
+    predictions = [
+        CuboidPrediction(
+            label=CAR_LABEL,
+            position=Point3D(1.0, 1.0, 1.0),
+            dimensions=Point3D(10, 10, 5),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+        CuboidPrediction(
+            label=CAR_LABEL,
+            position=Point3D(999, 999, 999),
+            dimensions=Point3D(8, 8, 6),
+            yaw=0.0,
+            reference_id="item_A",
+        ),
+        CuboidPrediction(
+            label=CAR_LABEL,
+            position=Point3D(250, 250, 250),
+            dimensions=Point3D(2, 2, 2),
+            yaw=0.0,
```

Comment:
to complete the test set let's also add another prediction/annotation combination with same size but yaw angle difference (e.g. 45 deg) as otherwise all the tests are with axis-aligned bounding boxes only taking into account translation